File: prueba-de-envios-archivos-con-rama-samuel/practica.py
import math

def practica(numero):
    # Los números menores que 2 no son primos
    if numero < 2:
        return False
    
    # Comprobar divisibilidad desde 2 hasta la raíz cuadrada del número
    for i in range(2, int(math.sqrt(numero)) + 1):
        if numero % i == 0:
            return False  # Si encontramos un divisor, el número no es primo
    
    return True  # Si no encontramos divisores, el número es primo

# Definir un número para comprobar
numero = 29  # Puedes cambiar este número para probar otros valores
es_primo = practica(numero)

if es_primo:
    print(f"{numero} es un número primo.")
else:
    print(f"{numero} no es un número primo.")

File: prueba-de-envios-archivos-con-rama-samuel/practica.java
public class practica {

    public static void main(String[] args) {
        int numero = 29; // Puedes cambiar este número para probar otros valores
        int numero_2 = 30;
        boolean esPrimo = esNumeroPrimo(numero);
        boolean esPrimo2 = esNumeroPrimo_Two(numero_2);

        if (esPrimo) {
            System.out.println(numero + " es un número primo.");
        } else {
            System.out.println(numero_2 + " no es un número primo.");
        }

        if (esPrimo2){
            System.out.println(numero_2 + "es un numero primo");
        } else{
            System.out.println(numero_2 + "no es un numero pirmo");
        }
    }

    public static boolean esNumeroPrimo(int numero) {
        // Los números menores que 2 no son primos
        if (numero < 2) {
            return false;
        }

        // Comprobar divisibilidad desde 2 hasta la raíz cuadrada del número
        for (int i = 2; i <= Math.sqrt(numero); i++) {
            if (numero % i == 0) {
                return false; // Si encontramos un divisor, el número no es primo
            }
 
         }

        return true; // Si no encontramos divisores, el número es primo

    }

    public static boolean esNumeroPrimo_Two(int numero_2){
        if (numero_2 <2) {
           return false;
        }
        for (int i =2; i <= Math.sqrt(numero_2); i++){
            if (numero_2 % i ==0){
            return false;
            }
        }
        return true;
    }
 
}
